sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -834,4 +834,21 @@ object IntervalUtils {
    * @return The period of months, not null
    */
   def monthsToPeriod(months: Int): Period = Period.ofMonths(months).normalized()
+
+  /**
+   * Converts an year-month interval as a number of months to its textual representation
+   * which conforms to the ANSI SQL standard.
+   *
+   * @param months The number of months, positive or negative
+   * @return Year-month interval string
+   */
+  def toYMIntervalString(months: Int): String = {
```

Comment:
Nit: let's user the full name here `toYearMonthIntervalString`

Reply:
The shortcut `YM` is defined by the SQL standard, see `6.13 <cast specification>`:
<img width="654" alt="Screenshot 2021-04-06 at 11 53 07" src="https://user-images.githubusercontent.com/1580697/113685241-b4761900-96ce-11eb-8b4f-53a0cbddcda7.png">

Even you used it in your PR https://github.com/apache/spark/pull/30260/files#diff-393a0ad195d66fb094f2dca3669c1317b7f57a6637f6e95241b57bc6550ce49aR1822 :-)

Reply:
renamed it